### ISSUE TITLE
Homework 4 (data visualization)

### ISSUE THREAD

**Issue by cmurray3** [IE-555/fall2023#12]
Homework 4 (data viz) is now posted:

- https://github.com/IE-555/fall2023/blob/main/Homework/HW_4_data_viz/HW_4_data_viz.pdf

If you have questions about the assignment, please pose them by replying to this issue below. Emailed homework questions will not be answered.

**Comment by thatonenerd2000** [comment]
For part 3 of the homework it states, `You may assume that the second column will only contain the following text strings:
\optimal", \genetic algorithm", \simulated annealing", and \tabu search".`

But all the data in the file is formatted like this `4,|simulated annealing|,22.944971` where there are pipes on both ends of the string. Are we supposed to remove/filter those?

**Comment by cmurray3** [comment]
> For part 3 of the homework it states, `You may assume that the second column will only contain the following text strings: \optimal", \genetic algorithm", \simulated annealing", and \tabu search".`
> 
> But all the data in the file is formatted like this `4,|simulated annealing|,22.944971` where there are pipes on both ends of the string. Are we supposed to remove/filter those?

Thanks for catching this!  I don't know how those `|` (pipe) characters got into that `.csv` file.  

I have updated the file (see https://github.com/IE-555/fall2023/blob/main/Homework/HW_4_data_viz/solution_data.csv), removing those extra characters.  There is nothing that you need to remove/filter now.

**Comment by cmurray3** [comment]
# Homework 4 (Data Visualization)

Dear IE 459/555 Students,

As we've discussed in class, we will be peer-evaluating Homework 4 (data visualization) in class on Monday, Nov. 6 and Wednesday, Nov. 8.  

**The link to create your GitHub repo for the assignment is here:**
- [https://classroom.github.com/a/KC63KwOC](https://classroom.github.com/a/KC63KwOC)

--- 

Some important info:
1. Upload your Jupyter notebook **before class** on Monday, Nov. 6.
    - Use `sample_hw4.ipynb` as a template/example 
    - Be sure to rename the Jupyter notebook, replacing `sample` with your UB userID.  For example, my Jupyter notebook would be renamed `cmurray3_hw4.ipynb`.
    - Remove `sample_hw4.ipynb` from the repo...there should only be **one** `.ipynb` file.  Otherwise, your repo will be cluttered with unnecessary files.
2. There is **no auto-grader** for this assignment.  Again, we will be peer-evaluating in class.  At the beginning of class I will assign you a partner and will explain how to give your partner access to your repo.
    - Make sure you show up on time so you can hear the instructions.
3. You must attend both classes (November 6th and 8th) to receive credit for this assignment.

**Comment by thatonenerd2000** [comment]
Do we upload the `.csv` file we used for part 2 too? My code opens the file to render the data but I have the link to the original file in the markdown.

**Comment by cmurray3** [comment]
> Do we upload the `.csv` file we used for part 2 too? My code opens the file to render the data but I have the link to the original file in the markdown.

Yes, please upload the .csv you used for Part 2. It’s good that you also provide the link in your documentation.
